File: backend/app/schemas/notification.py
from datetime import date
from typing import Literal

from pydantic import BaseModel


class LowStockItem(BaseModel):
    product_id: int
    product_name: str
    branch_id: int
    quantity: int
    threshold: int
    is_read: bool = False


class ExpiringItem(BaseModel):
    product_id: int
    product_name: str
    branch_id: int
    best_before_date: date
    is_read: bool = False


class NotificationsOut(BaseModel):
    low_stock: list[LowStockItem]
    expiring: list[ExpiringItem]


class NotificationReadIn(BaseModel):
    kind: Literal["low_stock", "expiring"]
    product_id: int
    branch_id: int

File: mobile/src/types/notification.ts
// backend/app/schemas/notification.py ile birebir eşleşir.
export interface LowStockItem {
  product_id: number;
  product_name: string;
  branch_id: number;
  quantity: number;
  threshold: number;
  is_read: boolean;
}

export interface ExpiringItem {
  product_id: number;
  product_name: string;
  branch_id: number;
  best_before_date: string;
  is_read: boolean;
}

export interface NotificationsOut {
  low_stock: LowStockItem[];
  expiring: ExpiringItem[];
}

export type NotificationKind = "low_stock" | "expiring";

export interface NotificationReadIn {
  kind: NotificationKind;
  product_id: number;
  branch_id: number;
}
